catroid/src/main/java/org/catrobat/catroid/merge/ConflictHelper.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.merge;
+
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.formulaeditor.UserList;
+import org.catrobat.catroid.formulaeditor.UserVariable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ConflictHelper {
+	public ConflictHelper() {
+	}
+
+	public boolean getGlobalVariableConflicts(List<UserVariable> project1GlobalVars, List<UserVariable> project2GlobalVars) {
+		List<UserVariable> conflicts = new ArrayList<>();
+		for (UserVariable project1GlobalVar : project1GlobalVars) {
+			for (UserVariable project2GlobalVar : project2GlobalVars) {
+				if (project1GlobalVar.getName().equals(project2GlobalVar.getName()) && !project1GlobalVar.getValue()
+						.equals(project2GlobalVar.getValue())) {
+					conflicts.add(project1GlobalVar);
+				}
+			}
+		}
+		return conflicts.isEmpty();
```

Comment:
You could optimize this in a way like don't saving an extra list with the conflicts added. This should may be implemented in a ticket to resolve tickets. Just to understand, you are creating a new list and afterwards you are just returning a property of this list. Think about something like (Pseudo Code):
```WHILE iterate_over_lists:
       IF conflict_found
               RETURN true;
       END IF
  END WHILE

   return false;```

Reply:
I did that on purpose. This ticket is a means to an end. Later on, I would need to resolve the conflicts so I save all the objects with conflicts for later use. There are two tradeoffs here, either space or speed. I did not want to repeat the process of looping over the lists to find conflicts in the future so I decided to save them instead.